src/protocol_splitter.cpp

```diff
@@ -516,16 +559,50 @@ void signal_handler(int signum)
 	running = false;
 }
 
-void serial_to_udp(pollfd *fds)
+void serial_to_udp(pollfd *fd_uart)
 {
 	while (running) {
-		if ((::poll(fds, sizeof(fds) / sizeof(fds[0]), 100) > 0) && (fds[0].revents & POLLIN)) {
-			// Start the timer for the pass-through activation on timeout at the first data poll
-			if (objects->serial->_timer_start == std::chrono::time_point<std::chrono::system_clock>()) {
-				objects->serial->_timer_start = std::chrono::system_clock::now();
+		const int ret = ::poll(fd_uart, sizeof(fd_uart) / sizeof(fd_uart[0]), 1000);
+
+		if (ret <= 0) {
+			// In case of a poll timeout or error, try to reopen the UART fd
+			//
+			// This is also done for timeouts as it was being verified for the case of
+			// the MAVLink passthrough, on some specific platforms, the UART poll was
+			// always timing out.
+			//
+			// @todo revisit the UART configs to understand why the timeout happens
+			if (ret == 0) {
+				printf("\033[1;33m[ protocol__splitter ]\tUART link: Poll timeout. ");
+
+			} else {
+				printf("\033[1;33m[ protocol__splitter ]\tUART link: Poll error (%d). ", ret);
 			}
 
-			objects->serial->read();
+			printf("Re-opening UART link and resetting fds...\033[0m\n");
+
+			std::unique_lock<std::mutex> uart_guard(uart_mtx);
```

Comment:
This needs to be using `socket_write_mtx`, which protects `_uart_fd` (and should be renamed `uart_mtx`)
```suggestion
			std::unique_lock<std::mutex> uart_guard(socket_write_mtx);
```

Reply:
Ah yes we are protecting the variable access, not the operation. Fixed.